bwilab_scripts/scripts/upload

```diff
@@ -2,56 +2,65 @@
 
 ## Upload saved bag files from the current directory.
 
-if [ "$#" -gt "1" -o "$1" = "-h" ]
-then    echo "usage: `basename $0` [option]"
+if [ "$#" -gt "2" ] || [ "$1" = "-h" ] || [ "$1" = "--help" ]
+then    echo "usage: $(basename $0) [option]"
         echo ""
         echo "Uploads files matching 'bwi_*.bag' from the current directory to the"
         echo "main BWI server. \$HOSTNAME should be the lowest-level domain name."
+        echo "By default, will delete the files after uploading"
         echo ""
-        echo "Supports at most one of these options:"
-        echo -e "\t-d\tdelete files uploaded successfully (default)"
-        echo -e "\t-h\tprint this help message"
-        echo -e "\t-k\tkeep uploaded files"
+        echo "Supports these options:"
+        echo -e "\t-h [--help]\tprint this help message"
+        echo -e "\t-p\tspecify the bag prefix (will upload nodes matching 'PREFIX_*.bag'"
+        echo -e "\t-e\tkeep the files after uploading"
         exit 1
 fi
-del_opt=${1:-"-d"}
-if [ "$del_opt" = "-d" ]
-then    del_opt="--remove-sent-files"
-else
-        if [ "$del_opt" = "-k" ]
-        then    del_opt=""
-        else    echo "invalid parameter: $del_opt (try -h for help)"
-                exit 2
-        fi
-fi
 
+# Default operations
+del_opt="--remove-sent-files"
+prefix="bwi"
+
+# Get parameters
+while getopts 'k:p:' opt; do
+    case $opt in
+        p)  prefix="$OPTARG" ;;
+        k)  del_opt=""       ;;
+        *)  exit 1           ;;
+    esac
+done
+
+suffix="_*.bag"
+glob=$prefix$suffix
+
+# Ensure the hostname is set
 if [ "$HOSTNAME" == "" ]
-then    echo '$HOSTNAME must be set for this operation'
+then    echo "$HOSTNAME must be set for this operation"
```

Comment:
This change causes the shell to attempt to resolve ``$HOSTNAME``, which is not defined. Using single quotes prints $HOSTNAME as part of the message.